docs(presets): update the tokyo-night preset

VictorPLopes opened the pull request:
#### Description
Update a preset: Tokyo Night

#### Motivation and Context
Replace the "Apple" nerd font icon with the user's OS (with the appropriate nerd font icons), ~~and enable the user module by default~~.
**Edit:** ~~also improve the gradient,~~ refactor colors to use a palette and update them to be consistent with the [current version of the source theme](https://github.com/enkia/tokyo-night-vscode-theme), while keeping the old colors as an alternate palette. Add support for more languages.

#### Screenshots (if appropriate):
Windows Terminal with Powershell:
![image](https://github.com/VictorPLopes/starship/blob/master/docs/.vuepress/public/presets/img/tokyo-night.png?raw=true)

#### How Has This Been Tested?
- [ ] I have tested using **MacOS**
- [ ] I have tested using **Linux**
- [x] I have tested using **Windows**

#### Checklist:
- [x] I have updated the documentation accordingly.
- ~~[ ] I have updated the tests accordingly.~~ **This PR has no code changes.**

VictorPLopes committed:
Update tokyo-night.toml

Improved preset. Replaces the apple icon with the OS module, with a nerd font icon for each OS. It also enables the user module by default.

VictorPLopes committed:
Add files via upload

VictorPLopes committed:
Delete tokyo-night.png

VictorPLopes committed:
Delete image.png

VictorPLopes committed:
Update tokyo-night

Updated screenshot of my improved preset

VictorPLopes committed:
Updated tokyo-night screenshot

Updated the tokyo-night screenshot to match the improved preset

VictorPLopes committed:
Update tokyo-night.md

Fixed the URLs

VictorPLopes committed:
Update tokyo-night.md

Fixed the URLs

snaeil commented:
@VictorPLopes Btw. the `Return to Presets` button points to https://starship.rs/presets/#pastel-powerline instead of https://starship.rs/presets/#tokyo-night
This could be fixed as well...

mickimnet commented on docs/presets/tokyo-night.md:
```diff
@@ -4,7 +4,7 @@
 This preset is inspired by [tokyo-night-vscode-theme](https://github.com/enkia/tokyo-night-vscode-theme).
 
-![Screenshot of Tokyo Night preset](/presets/img/tokyo-night.png)
+![Screenshot of Tokyo Night preset](/docs/.vuepress/public/presets/img/tokyo-night.png)
```
Please remove the unnecessary change.
```suggestion
![Screenshot of Tokyo Night preset](/public/presets/img/tokyo-night.png)
```

mickimnet commented on docs/presets/tokyo-night.md:
````diff
@@ -16,6 +16,6 @@ This preset is inspired by [tokyo-night-vscode-theme](https://github.com/enkia/t
 ```
 
-[Click to download TOML](/presets/toml/tokyo-night.toml)
+[Click to download TOML](/docs/.vuepress/public/presets/toml/tokyo-night.toml)
````
Please remove the unnecessary change.
```suggestion
[Click to download TOML](/presets/toml/tokyo-night.toml)
```

mickimnet commented on docs/presets/tokyo-night.md:
```diff
@@ -16,6 +16,6 @@ This preset is inspired by [tokyo-night-vscode-theme](https://github.com/enkia/t
+[Click to download TOML](/docs/.vuepress/public/presets/toml/tokyo-night.toml)
 
-<<< @/.vuepress/public/presets/toml/tokyo-night.toml
+<<< @/docs/.vuepress/public/presets/toml/tokyo-night.toml
```
Please use the following default.
```suggestion
<<< @/.vuepress/public/presets/toml/tokyo-night.toml
```

mickimnet commented on docs/.vuepress/public/presets/toml/tokyo-night.toml:
```diff
@@ -16,6 +20,52 @@ $time\
+style_user = "bg:#a3aed2 fg:#090c0c"
+style_root = "bg:#a3aed2 fg:#090c0c"
+format = '[$user ]($style)'
+disabled = false
```
In my perspective, I would prefer it, if the new default, respects the one from the original author.
Meaning, please use `disable = true` for the `[username]` module.

VictorPLopes replied:
Fixed.

mickimnet reviewed (changes requested):
Thank you very much for updating the `Tokyo Night` preset. I like that you added new modules, that fit the style of the original author. And as already commented, would prefer, if you would respect the intention, to not show the `username` module.

Additionally, as we now have the `[palettes]` module, and the majority of the colors do repeat, could you please refactor the colors to use the `[palettes]` module?

davidkna commented on docs/.vuepress/public/presets/toml/tokyo-night.toml:
```diff
@@ -37,18 +87,28 @@ format = '[[ $symbol $branch ](fg:#769ff0 bg:#394260)]($style)'
 
 [golang]
-symbol = ""
+symbol = "ﳑ"
```
Please don't use any obsolete nerd font icons.

VictorPLopes replied:
Fixed.

VictorPLopes commented:
@snaeil @davidkna @mickimnet thank you for the feedback, I will do the changes.

VictorPLopes committed:
Update docs/presets/tokyo-night.md

Fixed the default URL

Co-authored-by: Mick Hohmann <gh@m69.im.net>

VictorPLopes committed:
Update docs/presets/tokyo-night.md

Reverted unnecessary change.

Co-authored-by: Mick Hohmann <gh@m69.im.net>

VictorPLopes committed:
Update docs/presets/tokyo-night.md

Reverted unnecessary change.

Co-authored-by: Mick Hohmann <gh@m69.im.net>

VictorPLopes committed:
Update tokyo-night.toml

Implemented the suggested changes, replaced the "gradient" characters with a better color gradient, updated the colors to be consistent with the current version of the VSCode theme (https://github.com/enkia/tokyo-night-vscode-theme), implemented two palettes (one with the updated colors and another one with the old colors) and added support for more languages.

VictorPLopes commented on docs/.vuepress/public/presets/toml/tokyo-night.toml:
```diff
@@ -16,6 +20,52 @@ $time\
 \n$character"""
 
+# Disable the blank line at the start of the prompt
+add_newline = true
```
Fixed.

VictorPLopes commented:
> Thank you very much for updating the `Tokyo Night` preset. I like that you added new modules, that fit the style of the original author. And as already commented, would prefer, if you would respect the intention, to not show the `username` module.
> 
> Additionally, as we now have the `[palettes]` module, and the majority of the colors do repeat, could you please refactor the colors to use the `[palettes]` module?

Thank you very much for the feedback. I have reverted the `username` and refactored the colors to use the `[palettes]` module. I have also updated the colors to be consistent with the current version of the source theme and kept the old colors as an alt palette.

Updated colors (`tokyo_night` palette):
![image](https://github.com/starship/starship/assets/77900343/f57d3862-ff75-4a27-b297-1876ca76ad63)

Legacy colors (`toky_night_legacy` palette):
![image](https://github.com/starship/starship/assets/77900343/daa6cde5-9a09-4803-a5e3-dff053d5427b)

VictorPLopes committed:
Merge branch 'starship:master' into master

VictorPLopes committed:
Updated tokyo-night.png

VictorPLopes committed:
Update tokyo-night.md

VictorPLopes commented:
> 

Fixed

VictorPLopes commented:
> @VictorPLopes Btw. the `Return to Presets` button points to https://starship.rs/presets/#pastel-powerline instead of https://starship.rs/presets/#tokyo-night This could be fixed as well...

Fixed. Thank you.

VictorPLopes committed:
Update tokyo-night.toml

Removed Docker, which shouldn't be there

VictorPLopes committed:
Update tokyo-night.toml

Fixed # in front of color names

VictorPLopes committed:
Merge branch 'starship:master' into master

VictorPLopes committed:
Merge branch 'starship:master' into master

mickimnet commented on docs/.vuepress/public/presets/toml/tokyo-night.toml:
```diff
@@ -1,64 +1,205 @@
 $time\
-[ ](fg:#1d2230)\
+[ ](fg:black)\
 \n$character"""
```
Sorry, I totally for got the following suggestion
```suggestion
$line_break$character"""
```

The `[line_break]` module, is enabled by default. By using it, the user can disable it easily with `starship config line_break.disabled false`, if they so desire it.

mickimnet reviewed (commented):
Everything else looks good to me.

VictorPLopes committed:
Update docs/.vuepress/public/presets/toml/tokyo-night.toml

Replaced \n with the line_break module

Co-authored-by: Mick Hohmann <gh@m69.im.net>

VictorPLopes commented:
> Everything else looks good to me.

Great idea, I committed it, thank you.

mickimnet reviewed (approved):
LGTM.

davidkna commented on docs/.vuepress/public/presets/toml/tokyo-night.toml:
```diff
@@ -1,64 +1,205 @@
-[░▒▓](#a3aed2)\
-[  ](bg:#a3aed2 fg:#090c0c)\
-[](bg:#769ff0 fg:#a3aed2)\
+[ ](bg:blurple_darkest)\
```
I would prefer staying with `░▒▓` to avoid changing the preset too much.

VictorPLopes replied:
I reverted it like you requested. I do think it looks "worse" though, especially in some terminals with specific fonts, and that the colored rectangles do a better job of matching the original author's screenshot across different environments.

VictorPLopes committed:
Update tokyo-night.toml

Reverted gradient

VictorPLopes committed:
Update tokyo-night.toml

Disabled user module

VictorPLopes committed:
Add files via upload

VictorPLopes committed:
Update tokyo-night.toml

Changed the order of the [os] and [user] modules

VictorPLopes committed:
Merge branch 'starship:master' into master

GrumpyMetalGuy commented:
Hi, 

It still seems that the apple icon is used instead of (in my case) Linux. Can these conflicts be resolved so that it can be properly merged?

Thanks!

VictorPLopes commented:
> > > Hi,
> > > It still seems that the apple icon is used instead of (in my case) Linux. Can these conflicts be resolved so that it can be properly merged?
> > > Thanks!
> > 
> > 
> > Hello!
> > Which distro are you using at the moment?
> 
> I'm running Mint Cinnamon, and pulled the starship install down using the curl command. Let me know if you need any more info!

Hmm, this is very weird. I can't test it myself right now, but can you manually check if you're using the [preset from my for](https://github.com/VictorPLopes/starship/blob/7e4823a8ce16b04d9848310b811c24833f5c8540/docs/.vuepress/public/presets/toml/tokyo-night.toml)? Try manually downloading the TOML file and putting it into your presets folder.
